### ISSUE TITLE
Wrong behaviour with Get Requests

### ISSUE THREAD

**Issue by Rayv1** [furaiev/amazon-cognito-identity-dart-2#7]
@furaiev 

SignedRequests have still a Content-Type Header even if its a GET method. 

sig_v4.dart:

As first the Header is set to application/json as its default.
```
    if (headers['Content-Type'] == null) {
      headers['Content-Type'] = awsSigV4Client.defaultContentType;
    }
```

This If would remove the Header, if the Body is '' 
```
    if (body == '') {
      headers.remove('Content-Type');
    }
```

As as next the _generateAuthorization is triggered, which builds the Canonical Request and the Authorization Header. So if body is '', the Header is removed and the request is without correct headers, because as last, there is an additional IF which would add the Header if null.

```
    if (headers['Content-Type'] == null) {
      print("8");
      headers['Content-Type'] = awsSigV4Client.defaultContentType;
    }
```

Important is, that the Content-Type will never be removed. This is not a real problem if you work with the http package , because it simply sends the content-type with the request and therefore the SigV4 request is valid.

But there is a problem with dio, because it removes the content-type header from GET requests, because there is no content. 

I found a working Solution for now:
```
// Add the Content Type only if method is not GET
    if (headers['Content-Type'] == null **&& this.method != 'GET'**) {
      headers['Content-Type'] = awsSigV4Client.defaultContentType;
    }
```
Also removed following Code at the end of Constructor.
```
     if (headers['Content-Type'] == null) {
       print("setting");
      headers['Content-Type'] = awsSigV4Client.defaultContentType;
    }
```

**Comment by furaiev** [comment]
Thanks, I'll check it.

**Comment by furaiev** [comment]
Available as of **0.1.5+2**
